modules/core/src/main/scala/fs2/kafka/TransactionalKafkaProducer.scala

```diff
@@ -141,11 +142,26 @@ object TransactionalKafkaProducer {
         ): F[Chunk[(ProducerRecord[K, V], RecordMetadata)]] =
           withProducer.exclusiveAccess { (producer, blocking) =>
             blocking(producer.beginTransaction()).bracketCase { _ =>
-              val produce = records
-                .traverse(
-                  KafkaProducer.produceRecord(keySerializer, valueSerializer, producer, blocking)
-                )
-                .flatMap(_.sequence)
+              def produceRecords(produceRecordError: Option[Promise[Throwable]]) =
+                records
+                  .traverse(
+                    KafkaProducer.produceRecord(
+                      keySerializer,
+                      valueSerializer,
+                      producer,
+                      blocking,
+                      produceRecordError
+                    )
+                  )
+                  .flatMap(_.sequence)
+
+              val produce =
+                if (settings.producerSettings.failFastProduce)
+                  Async[F]
+                    .delay(Promise[Throwable]())
+                    .flatMap(produceRecordError => produceRecords(produceRecordError.some))
```

Comment:
A `Promise` is created but never listened (if I read correctly). Is that deliberate?

Reply:
The author of this PR is on holiday so I'll try to answer for him: it seems that the race that is present in modules/core/src/main/scala/fs2/kafka/KafkaProducer.scala, was not added here. It is my guess is that this was an oversight.

Reply:
yes, the race is missing.
I'll add it.

Reply:
good catch. It's added in commit 6442060eeb2aac91dcafc414e4d77df353f85c9f

I also added a test for it:
https://github.com/fd4s/fs2-kafka/blob/6442060eeb2aac91dcafc414e4d77df353f85c9f/modules/core/src/test/scala/fs2/kafka/TransactionalKafkaProducerSpec.scala#L479-L489

An interesting discovery is that `KafkaProducer.produce` and `TransactionalKafkaProducer.produce` behave differently. Specifically, `KafkaProducer.produce` defers the responsibility of waiting for callback resolutions to the user, as the F[F[_]] result. In contrast, `TransactionalKafkaProducer.produce` eagerly resolves the callbacks.

https://github.com/fd4s/fs2-kafka/blob/64a751e44d6a810d141e11beb8eef1ea8b532f60/modules/core/src/main/scala/fs2/kafka/KafkaProducer.scala#L45-L47

https://github.com/fd4s/fs2-kafka/blob/64a751e44d6a810d141e11beb8eef1ea8b532f60/modules/core/src/main/scala/fs2/kafka/TransactionalKafkaProducer.scala#L41-L43

https://github.com/fd4s/fs2-kafka/blob/64a751e44d6a810d141e11beb8eef1ea8b532f60/modules/core/src/main/scala/fs2/kafka/TransactionalKafkaProducer.scala#L144-L148

https://github.com/fd4s/fs2-kafka/blob/64a751e44d6a810d141e11beb8eef1ea8b532f60/modules/core/src/main/scala/fs2/kafka/KafkaProducer.scala#L178-L180

note the `flatMap` and `map` difference.